crates/driver/src/domain/competition/mod.rs

```diff
@@ -231,15 +237,15 @@ impl Competition {
 }
 
 /// Solution information revealed to the protocol by the driver before the
-/// settlement happens.
-/// Initially the calldata was planned to be revealed to the protocol after
-/// the \settle is called, but this would require protocol to wait for the
-/// settlement to be settled onchain before saving the solver competition data.
-/// This is unacceptable because saving the solver competition data could fail
-/// and the protocol would have inconsistent db state.
+/// settlement happens. Note that the calldata is only revealed once the
+/// protocol instructs the driver to reveal, and not before.
```

Comment:
Looks like this doc comment could be split to have the relevant information on `Solved` and `Revealed`.

Reply:
You are right. I've updated the docs to reflect the changes.